sdk/iot/hub/inc/az_iot_hub_client.h

```diff
@@ -182,7 +182,7 @@ typedef struct az_iot_hub_client_properties
   struct
   {
     az_span properties;
-    uint8_t* current_property;
+    uint32_t current_property_index;

```

Comment:
Indexes should be signed.

Reply:
I think it's safer and faster to use the pointer as it will automatically take the address size of the platform.

Reply:
The problem I ran into is it doesn't gel well with the `az_span_slice` api. There's another way to do this where I initialize a new span with the ptr from ^ that and then calculate length between the span length and pointer. It seems a little more complicated to me though.

Reply:
As to the point about signedness, the compiler didn't like pointer arithmetic resolving to signed integers.